Version 0.26.0 does not work on Windows due to incompatible jextract code

It seems the newly released version 0.26.0 does not work on Windows:
```java
java.lang.ExceptionInInitializerError
	at io.github.treesitter.jtreesitter.Language.<clinit>(Language.java:19)
	at ...
Caused by: java.lang.ClassCastException: class jdk.internal.foreign.layout.ValueLayouts$OfIntImpl cannot be cast to class java.lang.foreign.ValueLayout$OfLong (jdk.internal.foreign.layout.ValueLayouts$OfIntImpl and java.lang.foreign.ValueLayout$OfLong are in module java.base of loader 'bootstrap')
	at io.github.treesitter.jtreesitter.internal.TreeSitter$shared.<clinit>(TreeSitter$shared.java:30)
	... 4 more
```

The problem appears to be that for the release build jtreesitter assumes that the jextract-generated code for Linux also works for all other supported OS (macOS and Windows). However, due to https://bugs.openjdk.org/browse/CODETOOLS-7903923 (see also linked GitHub PR there) at least for Windows that is no longer the case.

Maybe this could be solved with the following patch for the generated `TreeSitter$shared.java`?
```diff
- public static final ValueLayout.OfLong C_LONG = (ValueLayout.OfLong) Linker.nativeLinker().canonicalLayouts().get("long");
+ public static final ValueLayout.OfLong C_LONG = C_LONG_LONG;
```

For Linux I think this is correct? For Windows this is technically incorrect but it seems for all function signatures the jextract generated code uses `C_LONG_LONG` instead of `C_LONG` on Windows, so effectively this patch here would account for that? Please double-check if possible though.
You can for example modify the CI workflow for testing to upload the jextract-generated code as artifact (as [done here](https://github.com/Marcono1234/java-tree-sitter/actions/runs/20471926174#artifacts)).

The reason why the Windows GitHub CI workflow is not failing is probably because it runs jextract itself and therefore generates Windows-compatible code. Maybe it would be better if the CI workflow ran jextract only for Linux, uploaded the results as artifacts and then reused them for all OS, to match the behavior of the release workflow?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -86,7 +86,7 @@ jobs:
           printf '%s/jextract-25/bin\n' "$RUNNER_TOOL_CACHE" >> "$GITHUB_PATH"
         env:
           # NOTE: keep this in sync with deploy, docs
-          JEXTRACT_URL: https://download.java.net/java/early_access/jextract/25/1/openjdk-25-jextract+1-1
+          JEXTRACT_URL: https://download.java.net/java/early_access/jextract/25/2/openjdk-25-jextract+2-4
       - name: Run tests
         run: mvn --no-transfer-progress test
       - name: Patch SpotBugs SARIF report

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -43,15 +43,18 @@ jobs:
           cd tree-sitter-java
           tree-sitter generate --abi=15
           make all install PREFIX="$RUNNER_TOOL_CACHE/tree-sitter/lib"
-      - name: Set up jextract
+      - name: Generate code with jextract
         run: |-
           curl -LSs '${{env.JEXTRACT_URL}}_linux-x64_bin.tar.gz' | tar xzf - -C "$RUNNER_TOOL_CACHE"
           printf '%s/jextract-25/bin\n' "$RUNNER_TOOL_CACHE" >> "$GITHUB_PATH"
+          mvn --no-transfer-progress antrun:run@jextract
+          sed -i '${{env.TS_SHARED}}' -e '/C_LONG =/s/"long"/"long long"/'
         env:
           # NOTE: keep this in sync with ci, docs
           JEXTRACT_URL: https://download.java.net/java/early_access/jextract/25/1/openjdk-25-jextract+1-1
+          TS_SHARED: target/generated-sources/jextract/io/github/treesitter/jtreesitter/internal/TreeSitter$shared.java
       - name: Deploy to Maven Central
-        run: mvn --no-transfer-progress deploy -Dspotbugs.skip=true
+        run: mvn --no-transfer-progress deploy -Dspotbugs.skip=true -Djextract.skip=true
         env:
           SONATYPE_USERNAME: ${{secrets.SONATYPE_USERNAME}}
           SONATYPE_TOKEN: ${{secrets.SONATYPE_TOKEN}}

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -51,7 +51,7 @@ jobs:
           sed -i '${{env.TS_SHARED}}' -e '/C_LONG =/s/"long"/"long long"/'
         env:
           # NOTE: keep this in sync with ci, docs
-          JEXTRACT_URL: https://download.java.net/java/early_access/jextract/25/1/openjdk-25-jextract+1-1
+          JEXTRACT_URL: https://download.java.net/java/early_access/jextract/25/2/openjdk-25-jextract+2-4
           TS_SHARED: target/generated-sources/jextract/io/github/treesitter/jtreesitter/internal/TreeSitter$shared.java
       - name: Deploy to Maven Central
         run: mvn --no-transfer-progress deploy -Dspotbugs.skip=true -Djextract.skip=true

--- a/.github/workflows/docs.yml
+++ b/.github/workflows/docs.yml
@@ -55,7 +55,7 @@ jobs:
           printf '%s/jextract-25/bin\n' "$RUNNER_TOOL_CACHE" >> "$GITHUB_PATH"
         env:
           # NOTE: keep this in sync with ci, deploy
-          JEXTRACT_URL: https://download.java.net/java/early_access/jextract/25/1/openjdk-25-jextract+1-1
+          JEXTRACT_URL: https://download.java.net/java/early_access/jextract/25/2/openjdk-25-jextract+2-4
       - name: Build javadoc
         run: mvn --no-transfer-progress javadoc:javadoc antrun:run@fix-javadoc
       - name: Upload pages artifact

--- a/src/main/java/io/github/treesitter/jtreesitter/internal/ChainedLibraryLookup.java
+++ b/src/main/java/io/github/treesitter/jtreesitter/internal/ChainedLibraryLookup.java
@@ -21,7 +21,8 @@ final class ChainedLibraryLookup implements NativeLibraryLookup {
             lookup = lookup.or(libraryLookup.get(arena));
         }
 
-        return lookup.or((name) -> findLibrary(arena).find(name)).or(Linker.nativeLinker().defaultLookup());
+        return lookup.or((name) -> findLibrary(arena).find(name))
+                .or(Linker.nativeLinker().defaultLookup());
     }
 
     private static SymbolLookup findLibrary(Arena arena) {

--- a/src/main/java/io/github/treesitter/jtreesitter/package-info.java
+++ b/src/main/java/io/github/treesitter/jtreesitter/package-info.java
@@ -5,7 +5,7 @@
  *
  * <ul>
  * <li>
- *     JDK 22 (for the
+ *     JDK 23+ (for the
  *     <a href="https://docs.oracle.com/en/java/javase/22/core/foreign-function-and-memory-api.html">
  *         Foreign Function and Memory API
  *     </a>)

